CMakeLists.txt

```diff
@@ -18,18 +18,23 @@ project(ROOT)
 # output to use the default C locale which is more or less identical on all systems.
 set(ENV{LANG} C)
 
-#---Set pathes where to put the libraries, executables and headers------------------------------
+#---Set paths where to put the libraries, executables and headers------------------------------
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 # Before setting ROOTSYS, make sure that the environment isn't polluted by a different
 # ROOT build. This is significant e,g. for roottest, which will otherwise have libraries
 # of a different ROOT build available / visible / reachable.
-if($ENV{ROOTSYS})
-  string(REPLACE "$ENV{ROOTSYS}/bin" "" ENV{PATH}, $ENV{PATH})
-  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV{LD_LIBRARY_PATH} $ENV{LD_LIBRARY_PATH})
-  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV{PYTHONPATH} $ENV{PYTHONPATH})
+if(NOT $ENV{ROOTSYS} STREQUAL "")
+  string(REPLACE "$ENV{ROOTSYS}/bin" "" ENV_PATH "$ENV{PATH}")
+  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV_LD_LIBRARY_PATH "$ENV{LD_LIBRARY_PATH}")
+  string(REPLACE "$ENV{ROOTSYS}/lib" "" ENV_PYTHONPATH $ENV{PYTHONPATH})
```

Comment:
Will this ever be `$ENV{ROOTSYS}/lib64`, leaving a `64` in the `PATH`?

Reply:
Answer: Not if you source `thisroot.sh`, but with `-Dgnuinstall=ON -DCMAKE_INSTALL_LIBDIR=lib64`, `thisroot.sh` won't work it seems.

Reply:
But if you do this then there shouldn't be any `thisroot.sh`? Anyway - is this something that must be fixed for this PR? ;-)

Reply:
No, this is not something to be fixed in this PR. It's just that `thisroot.sh` is installed with `make install`, so in principle that could happen when sourcing after installing. But I don't think anyone is actually customizing ROOT's installation dirs, so it's something that can be fixed as part of [ROOT-8589](https://sft.its.cern.ch/jira/browse/ROOT-8589).